snuba/datasets/entities/transactions.py

```diff
@@ -85,21 +98,119 @@
     ],
 )
 
+SAMPLE_RATE = 0.1
+
+
+class SampledSimplePipelineBuilder(SimplePipelineBuilder):
+    def build_planner(
+        self, query: LogicalQuery, settings: RequestSettings,
+    ) -> EntityQueryPlanner:
+        new_query = deepcopy(query)
+        new_query.set_sample(SAMPLE_RATE)
+        return super().build_planner(new_query, settings)
+
 
 class BaseTransactionsEntity(Entity, ABC):
     def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         storage = get_writable_storage(StorageKey.TRANSACTIONS)
         schema = storage.get_table_writer().get_schema()
 
+        pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=storage,
+                mappers=transaction_translator
+                if custom_mappers is None
+                else transaction_translator.concat(custom_mappers),
+            ),
+        )
+        sampled_pipeline_builder = SampledSimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=storage,
+                mappers=transaction_translator
+                if custom_mappers is None
+                else transaction_translator.concat(custom_mappers),
+            ),
+        )
+
+        def selector_func(query: LogicalQuery, referrer: str) -> Tuple[str, List[str]]:
+            if referrer == "tagstore.__get_tag_keys":
+                condition = query.get_condition()
+                assert condition is not None
+                project_literals = get_literals_for_column_condition(
+                    "project_id", int, condition
+                )
+                values = set(lit.value for lit in project_literals)
+                if 1 in values:
```

Comment:
The project id of our internal project should not be hard coded in the public source.
Please add it as a prod only setting or as a config value.